src/Jsonyte/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -182,6 +185,18 @@ private IEnumerable<JsonMemberInfo> GetFields(Type type, JsonSerializerOptions o
             }
         }
 
+        private IEnumerable<JsonMemberInfo> GetInfoFromAttributes(Type type, JsonSerializerOptions options)
+        {
+            var attributes = type.GetCustomAttributes();
```

Comment:
You can use the generic type here to only get the attributes of the type we want, which will remove the need for a foreach.

eg. `type.GetCustomAttributes<JsonApiResourceAttribute>().Select(x => new MemberInfo blah())`